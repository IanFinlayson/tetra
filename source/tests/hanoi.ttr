global moves = 0

def moveDiscs(n int, from int, to int, temp int):
    if n > 0:
        moveDiscs(n - 1, from, temp, to)
        print("Move a disc from ", from, " to ", to, "\n")
        moves += 1
        moveDiscs(n - 1, temp, to, from)


def main():
    discs = 5
    moveDiscs(discs, 1, 3, 2)
    print("Done in ", moves, " moves.")

