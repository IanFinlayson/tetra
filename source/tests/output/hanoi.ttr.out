Move a disc from 1 to 3
Move a disc from 1 to 2
Move a disc from 3 to 2
Move a disc from 1 to 3
Move a disc from 2 to 1
Move a disc from 2 to 3
Move a disc from 1 to 3
Move a disc from 1 to 2
Move a disc from 3 to 2
Move a disc from 3 to 1
Move a disc from 2 to 1
Move a disc from 3 to 2
Move a disc from 1 to 3
Move a disc from 1 to 2
Move a disc from 3 to 2
Move a disc from 1 to 3
Move a disc from 2 to 1
Move a disc from 2 to 3
Move a disc from 1 to 3
Move a disc from 2 to 1
Move a disc from 3 to 2
Move a disc from 3 to 1
Move a disc from 2 to 1
Move a disc from 2 to 3
Move a disc from 1 to 3
Move a disc from 1 to 2
Move a disc from 3 to 2
Move a disc from 1 to 3
Move a disc from 2 to 1
Move a disc from 2 to 3
Move a disc from 1 to 3
Done in 31 moves.
